packaging/installer/netdata-updater.sh

```diff
@@ -48,16 +48,18 @@ else
   INTERACTIVE=1
 fi
 
+script_name=$(basename "$0")
```

Comment:
```suggestion
if [ -n "${script_source}" ]; then
  script_name="$(basename "${script_source}")"
else
  script_name="netdata-updater.sh"
fi
```

`$0` does not consistently behave as you might expect, and we already have some logic to handle this sanely, so it’s probably better to just use that here, and then fall back to a fixed name so we have a consistent name in all cases.

Reply:
Can you give an example when `basename "$0"` doesn't work for the netdata-updater script?

Reply:
Sourcing the script for example inherits the `$0` of the script sourcing it.

There are a couple of other odd edge cases as well.

The other thing here though is that this, combined with how we compute `${script_source}`, means we will consistently end up with a script name similar to `netdata-updater.sh`, which is kind of important for what we’re using this for (there is no point logging the name if we can end up with random, meaningless names).

Reply:
> Sourcing the script for example inherits the $0 of the script sourcing it.

Do we source netdata-updater?

> if we can end up with random, meaningless names

I agree with that, but so far I haven't seen any real example of how we can end up in that situation.

Reply:
We don’t invoke it ourselves in any way that could realistically cause this, but I’d rather have insurance against possible future changes and theoretical users making questionable decisions.